src/app/components/game-components/rectangular-game-component/RectangularGameComponent.ts

```diff
@@ -18,17 +19,32 @@ export abstract class RectangularGameComponent<R extends Rules<M, S, L>,
 
     public board: Table<P>;
 
+    public getWidth(): number {
+        return this.board[0].length;
+    }
+
+    public getHeight(): number {
+        return this.board.length;
+    }
+
     public getViewBox(): string {
-        return (-0.5 * this.STROKE_WIDTH) + ' ' +
-            (-0.5 * this.STROKE_WIDTH) + ' ' +
-            (this.board.length * this.SPACE_SIZE + this.STROKE_WIDTH) + ' ' +
-            (this.board[0].length * this.SPACE_SIZE + this.STROKE_WIDTH);
+        const left: number = -0.5 * this.STROKE_WIDTH;
+        const up: number = -0.5 * this.STROKE_WIDTH;
+        const right: number = this.getWidth() * this.SPACE_SIZE + this.STROKE_WIDTH;
+        const down: number = this.getHeight() * this.SPACE_SIZE + this.STROKE_WIDTH;
```

Comment:
c'est width et height !

Reply:
le côté droit de la viewbox c'est width*multiple, le côté bas c'est height*multiple, oui, mais donc "right" et "down" sont adapté comme noms

Reply:
```
The value of the viewBox attribute is a list of four numbers: min-x, min-y, width and height.
````
Selon developer.mozilla.org (et vu ailleurs aussi)

Ah mais en fait là t'appelle fromLimist qui en fait devrais plutôt reçevoir, pour la classe, un param de type
{ minX, maxX, minY, maxY } non ?

Reply:
ps: du coup c'pas plutôt un truc du genre
{ 0, 0, width * multiple, height * multiple } . add (un demistroke dans chaque direction) ?
Là tu asymétrise l'épaisseur du stroke !

Reply:
abawi, maintenant c'est symétrisé et plus lisible ! Les paramètres de type `{ foo, bar }`, pas fan du tout ! sauf pour des configs ou tu veux du par défaut et pouvoir spécifier seulement un sous-ensemble des paramètres

Reply:
btw: fixé la viewbox de tafl que j'avions cassé

Reply:
juste to be safe, je laisse un commentaire car là elle a encore l'upper half stroke qui manque